### PR TITLE
#895 org.junit.Test and expected Exceptions replaced

### PR DESCRIPTION
**Pull request by marceloamadeu**
For #895 org.junit.Test and expected Exceptions replaced

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#961](https://github.com/yegor256/takes/issues/961) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#961](https://github.com/yegor256/takes/issues/961) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C3FFK3YAY/692e4b4f-2b11-4411-9667-b7c04ec54763); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Job `gh:yegor256/takes#961` already assigned to @fabriciofx, can't assign to @fabriciofx

**Comment by marceloamadeu** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/takes/issues/961#issuecomment-462327774))

@marceloamadeu This is what I know about this job in [C3FFK3YAY](https://www.0crat.com/p/C3FFK3YAY), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#961](https://github.com/yegor256/takes/issues/961) is [in scope](http://www.zerocracy.com/policy.html#14) for 4days
  * The role is `REV`
  * The job is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) for [4days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @marceloamadeu[/z](https://www.0crat.com/u/marceloamadeu): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C3FFK3YAY?q=%7Bjob%3A%27gh%3Ayegor256%2Ftakes%23961%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by marceloamadeu** [comment]
@fabriciofx ping

**Comment by marceloamadeu** [comment]
@paulodamaso please take a look at this

**Comment by paulodamaso** [comment]
@fabriciofx ping

**Comment by paulodamaso** [comment]
@0crat assign me

File: src/test/java/org/takes/facets/auth/codecs/CcStrictTest.java [hunk 1]
```diff
@@ -39,24 +40,33 @@
...
-     *  all expected Exceptions by Assertions.assertThrows. Check
-     *  org.takes.facets.auth.codecs.CcAesTest or
-     *  org.takes.facets.auth.codecs.CcSaltedTest as examples.
      */
-    @Test(expected = DecodingException.class)
+    @Test
     public void blocksEmptyUrn() throws IOException {
-        new CcStrict(new CcPlain()).encode(new Identity.Simple(""));
+        Assertions.assertThrows(
+            DecodingException.class,
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/codecs/CcStrictTest.java]: @marceloamadeu Please add a message informing the reason of the assertion failure

File: src/test/java/org/takes/facets/auth/codecs/CcStrictTest.java [hunk 2]
```diff
@@ -39,24 +40,33 @@
...
     /**
      * CcStrict can block invalid urn.
      * @throws IOException If some problem inside
      */
-    @Test(expected = DecodingException.class)
+    @Test
     public void blocksInvalidUrn() throws IOException {
-        new CcStrict(new CcPlain()).decode("u%3Atest%3A9".getBytes());
+        Assertions.assertThrows(
+            DecodingException.class,
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/auth/codecs/CcStrictTest.java]: @marceloamadeu Here too, let's inform why the assertion failed

File: src/test/java/org/takes/facets/cookies/RsWithCookieTest.java [hunk 3]
```diff
@@ -101,17 +102,29 @@ public void addsMultipleCookies() throws IOException {
     /**
      * RsWithCookie can reject invalid cookie name.
      */
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void rejectsInvalidName() {
-        new RsWithCookie(new RsEmpty(), "f oo", "works");
+        Assertions.assertThrows(
+            IllegalArgumentException.class,
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/cookies/RsWithCookieTest.java]: @marceloamadeu Here again, put some message to inform the failure reason

File: src/test/java/org/takes/facets/cookies/RsWithCookieTest.java [hunk 4]
```diff
@@ -101,17 +102,29 @@ public void addsMultipleCookies() throws IOException {
...
 
     /**
      * RsWithCookie can reject invalid cookie value.
      */
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void rejectsInvalidValue() {
-        new RsWithCookie(new RsEmpty(), "cookiename", "wo\"rks");
+        Assertions.assertThrows(
+            IllegalArgumentException.class,
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/cookies/RsWithCookieTest.java]: @marceloamadeu Here too

File: src/test/java/org/takes/facets/ret/RsReturnTest.java [hunk 5]
```diff
@@ -65,8 +66,15 @@ public void addsCookieToResponse() throws IOException {
      * RsReturn can reject invalid location.
      * @throws IOException If some problem inside
      */
-    @Test(expected = IOException.class)
+    @Test
     public void rejectsInvalidLocation() throws IOException {
-        new RsReturn(new RsEmpty(), "http://www.netbout.com/,PsCookie=");
+        Assertions.assertThrows(
+            IOException.class,
+            () -> {
```
Comment (paulodamaso) [inline src/test/java/org/takes/facets/ret/RsReturnTest.java]: @marceloamadeu Again, here

File: src/test/java/org/takes/rq/RqHrefTest.java [hunk 6]
```diff
@@ -105,29 +106,39 @@ public void parsesHttpQueryWithoutHost() throws IOException {
...
      */
-    @Test(expected = HttpException.class)
+    @Test
     public void failsOnAbsentRequestLine() throws IOException {
-        new RqHref.Base(
-            new RqSimple(Collections.<String>emptyList(), null)
-        ).href();
+        Assertions.assertThrows(
+            HttpException.class,
+            () -> {
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqHrefTest.java]: @marceloamadeu Here too, put the message

File: src/test/java/org/takes/rq/RqHrefTest.java [hunk 7]
```diff
@@ -105,29 +106,39 @@ public void parsesHttpQueryWithoutHost() throws IOException {
...
-                ),
-                ""
-            )
-        ).href();
+        Assertions.assertThrows(
+            HttpException.class,
+            () -> {
+                new RqHref.Base(
+                    new RqFake(
+                        Arrays.asList(
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqHrefTest.java]: @marceloamadeu Use cactoos `ListOf` here

File: src/test/java/org/takes/rq/RqHrefTest.java [hunk 8]
```diff
@@ -105,29 +106,39 @@ public void parsesHttpQueryWithoutHost() throws IOException {
...
-                    "GIVE/contacts",
-                    "Host: 2.example.com"
-                ),
-                ""
-            )
-        ).href();
+        Assertions.assertThrows(
+            HttpException.class,
+            () -> {
+                new RqHref.Base(
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqHrefTest.java]: @marceloamadeu Put the assertion failure reason message here

File: src/test/java/org/takes/rq/RqLiveTest.java [hunk 9]
```diff
@@ -114,25 +115,35 @@ public void supportMultiLineHeadersWithColon() throws IOException {
...
      */
-    @Test(expected = IOException.class)
+    @Test
     public void failsOnBrokenHttpRequest() throws IOException {
-        new RqLive(
-            new ByteArrayInputStream(
-                "GET /test HTTP/1.1\r\nHost: \u20ac".getBytes()
-            )
+        Assertions.assertThrows(
+            IOException.class,
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqLiveTest.java]: @marceloamadeu Here too, put the assertion failure reason message

File: src/test/java/org/takes/rq/RqLiveTest.java [hunk 10]
```diff
@@ -114,25 +115,35 @@ public void supportMultiLineHeadersWithColon() throws IOException {
...
-    @Test(expected = IOException.class)
+    @Test
     public void failsOnInvalidCrLfInRequest() throws IOException {
-        new RqLive(
-            new ByteArrayInputStream(
-                "GET /test HTTP/1.1\rHost: localhost".getBytes()
-            )
+        Assertions.assertThrows(
+            IOException.class,
+            () -> {
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqLiveTest.java]: @marceloamadeu Here too

File: src/test/java/org/takes/rq/RqRequestLineTest.java [hunk 11]
```diff
@@ -44,11 +46,16 @@
...
      * @throws IOException If some problem inside
      */
-    @Test(expected = HttpException.class)
+    @Test
     public void failsOnAbsentRequestLine() throws IOException {
-        new RqRequestLine.Base(
-            new RqSimple(Collections.<String>emptyList(), null)
-        ).header();
+        Assertions.assertThrows(
+            DecodingException.class,
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqRequestLineTest.java]: @marceloamadeu Here too

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/takes/issues/961#issuecomment-465778942))

@paulodamaso @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) not enough funds available in the project, can't set budget of job [#961](https://github.com/yegor256/takes/issues/961), see [§21](http://www.zerocracy.com/policy.html#21); @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) will get no money on completion; in order to fix that, add funds to the project and assign the job again

File: src/test/java/org/takes/rq/RqRequestLineTest.java [hunk 12]
```diff
@@ -57,17 +64,22 @@ public void failsOnAbsentRequestLine() throws IOException {
...
-                ),
-                ""
-            )
-        ).header();
+        Assertions.assertThrows(
+            HttpException.class,
+            () -> {
+                new RqRequestLine.Base(
+                    new RqFake(
+                        Arrays.asList(
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqRequestLineTest.java]: @marceloamadeu Use `ListOf`

File: src/test/java/org/takes/rq/RqRequestLineTest.java [hunk 13]
```diff
@@ -57,17 +64,22 @@ public void failsOnAbsentRequestLine() throws IOException {
...
-            new RqFake(
-                Arrays.asList(
-                    "GIVE/contacts2",
-                    "Host: 1.example.com"
-                ),
-                ""
-            )
-        ).header();
+        Assertions.assertThrows(
+            HttpException.class,
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqRequestLineTest.java]: @marceloamadeu Put the message in the assertion

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

File: src/test/java/org/takes/rq/RqRequestLineTest.java [hunk 14]
```diff
@@ -99,11 +111,16 @@ public void extractsParams() throws IOException {
...
      */
-    @Test(expected = HttpException.class)
+    @Test
     public void failsOnAbsentRequestLineToken() throws IOException {
-        new RqRequestLine.Base(
-            new RqSimple(Collections.<String>emptyList(), null)
-        ).method();
+        Assertions.assertThrows(
+            HttpException.class,
+            () -> {
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqRequestLineTest.java]: @marceloamadeu Put the message in assertion

File: src/test/java/org/takes/rq/RqRequestLineTest.java [hunk 15]
```diff
@@ -99,11 +111,16 @@ public void extractsParams() throws IOException {
...
+                new RqRequestLine.Base(
+                    new RqSimple(Collections.<String>emptyList(), null)
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqRequestLineTest.java]: @marceloamadeu Use `ListOf` instead `Collections.<String>emptyList()` and remove this ugly null

File: src/test/java/org/takes/rq/RqRequestLineTest.java [hunk 16]
```diff
@@ -112,17 +129,22 @@ public void failsOnAbsentRequestLineToken() throws IOException {
...
-                ),
-                ""
-            )
-        ).method();
+        Assertions.assertThrows(
+            HttpException.class,
+            () -> {
+                new RqRequestLine.Base(
+                    new RqFake(
+                        Arrays.asList(
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/RqRequestLineTest.java]: @marceloamadeu Use `ListOf` and fix the assertion message

File: src/test/java/org/takes/rq/multipart/RqMtFakeTest.java [hunk 17]
```diff
@@ -82,6 +82,10 @@ public void throwsExceptionOnNoNameAtContentDispositionHeader()
      * RqMtFake can throw exception on no boundary
      * at Content-Type header.
      * @throws Exception if there is some problem inside
+     * @todo #895:30min replace all org.junit.Test occurrences in the project
```
Comment (paulodamaso) [inline src/test/java/org/takes/rq/multipart/RqMtFakeTest.java]: @marceloamadeu `Replace`

**Comment by paulodamaso** [comment]
@marceloamadeu ping

**Comment by 0crat** [comment]
Job [#961](https://github.com/yegor256/takes/issues/961) is not in the agenda of @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), can't retrieve data and time of add

**Comment by 0crat** [comment]
Code review was too long (341 days), architects (@paulodamaso) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Pull request [#961](https://github.com/yegor256/takes/issues/961) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Pull request closed without merging**